src/main/java/mate/academy/springboot/swagger/controller/ProductController.java

```diff
@@ -0,0 +1,117 @@
+import mate.academy.springboot.swagger.model.Product;
+import mate.academy.springboot.swagger.model.dto.request.ProductRequestDto;
+import mate.academy.springboot.swagger.model.dto.response.ProductResponseDto;
+import mate.academy.springboot.swagger.service.ProductService;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.domain.Sort;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/products")
+public class ProductController {
+    private final ProductService productService;
+    private final ProductRequestMapper productRequestMapper;
+    private final ProductResponseMapper productResponseMapper;
+
+    public ProductController(ProductService productService,
+                             ProductRequestMapper productRequestMapper,
+                             ProductResponseMapper productResponseMapper) {
+        this.productService = productService;
+        this.productRequestMapper = productRequestMapper;
+        this.productResponseMapper = productResponseMapper;
+    }
+
+    @PostMapping("/")
+    public ProductResponseDto create(@RequestBody ProductRequestDto dto) {
+        Product product = productService.create(productRequestMapper.fromDto(dto));
+        return productResponseMapper.toDto(product);
+    }
+
+    @GetMapping("/{id}")
+    public ProductResponseDto get(@PathVariable Long id) {
+        Product product = productService.get(id);
+        return productResponseMapper.toDto(product);
+    }
+
+    @PutMapping("/{id}")
+    public ProductResponseDto update(@PathVariable Long id, @RequestBody ProductRequestDto dto) {
+        Product product = productService.get(id);
+        product.setTitle(dto.getTitle());
+        product.setPrice(dto.getPrice());
+        productService.update(product);
+        return productResponseMapper.toDto(product);
+    }
+
+    @DeleteMapping("/{id}")
+    public ProductResponseDto delete(@PathVariable Long id) {
+        Product product = productService.get(id);
+        productService.delete(product);
+        return productResponseMapper.toDto(product);
+    }
+
+    @GetMapping
+    public List<ProductResponseDto> findAll(@RequestParam (defaultValue = "5") Integer count,
+                                            @RequestParam (defaultValue = "5") Integer page,
+                                            @RequestParam (defaultValue = "id") String sortBy) {
+
```

Comment:
Should that much logic be in the controller?

Reply:
According to  video/1960 3:40  it seems like possible